src/applications/common/schemaform/fields/ArrayField.jsx

```diff
@@ -192,14 +192,22 @@ export default class ArrayField extends React.Component {
     const DescriptionField = typeof description === 'function'
       ? uiSchema['ui:description']
       : null;
-    const isReviewMode = uiSchema['ui:options'].reviewMode;
+    const {
+      reviewMode: isReviewMode,
+      showLastItemInViewMode: viewMode
+    } = uiOptions;
     const hasTitleOrDescription = (!!title && !hideTitle) || !!description;
 
     // if we have form data, use that, otherwise use an array with a single default object
     const items = (formData && formData.length)
       ? formData
       : [getDefaultFormState(schema, undefined, registry.definitions)];
-
+    const lastItem = items[items.length - 1];
```

Comment:
For some reason, I'm having a hard time follow this added chunk. @cehsu Can you explain what this doing?

Reply:
This is checking that the last time (the item this change effects), is valid, to determine whether the showlastItemInViewMode configuration should be applied or not (if it is not valid).

Reply:
Would it make sense to add some comments here to clarify?